CMakeLists.txt

```diff
@@ -0,0 +1,356 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(coap VERSION 4.3.0 LANGUAGES CXX C)
```

Comment:
I know I suggested that this could be coap instead of libcoap to fix library naming issues, but as the library names actually include the encryption type I have suggested a fix for this in the add_lbrary() section and so I think that this should revert back to libcoap (which will affect some of your changes from "coap" to "${PROJECT_NAME}".  Sorry.

See https://github.com/obgm/libcoap/pull/151